lib/adapters/file/write.js

```diff
@@ -0,0 +1,85 @@
+var _ = require('underscore');
+var Juttle = require('../../runtime/index').Juttle;
+var fs = require('fs');
+
+var Write = Juttle.proc.sink.extend({
+    procName: 'write-file',
+    initialize: function(options, params) {
+        var allowed_options = ['file', 'limit'];
+        var unknown = _.difference(_.keys(options), allowed_options);
+        if (unknown.length > 0) {
+            throw this.compile_error('RT-UNKNOWN-OPTION-ERROR', {
+                proc: 'write file',
+                option: unknown[0]
+            });
+        }
+
+        if (!_.has(options, 'file')) {
+            throw this.compile_error('RT-MISSING-OPTION-ERROR', {
+                proc: 'write file',
+                option: "file"
+            });
+        }
+
+        this.limit = options.limit ? options.limit : 100000;
+        this.filename = options.file;
+        this.queue = [];
+    },
+
+    start: function() {
+        this.flush();
+    },
+
+    process: function(points) {
+        this.logger.debug('process', points);
+        this.queue = this.queue.concat(points);
+    },
+
+    flush: function() {
+        if (this.queue.length !== 0) {
+            var data = [];
+            var points = [];
+
+            try {
+                data = fs.readFileSync(this.filename, 'utf8');
+                points = JSON.parse(data);
+            } catch (err) {
+                if (err.code !== 'ENOENT') {
+                    this.logger.error('error reading file:', err.toString());
+                }
+            }
+
+            var space_left = this.limit - points.length;
```

Comment:
There's an edge case here - if there were already some points in the file, say 200k (e.g. because we're appending points to an existing file), and limit is set to 100k, space_left will be negative. This will cause the last 100k points to be added again. `space_left` minimum should be capped at 0.